app/forms/delayed_micropost_form.rb

```diff
@@ -0,0 +1,31 @@
+class DelayedMicropostForm
+  include ActiveModel::Model
+
+  validates :user, presence: true
+  validates :content, presence: true, length: { maximum: 140 }
+
+  attr_accessor :content, :is_delayed, :pub_date
+  attr_reader :user
+
+  def initialize(params)
+    @user = params[:user]
+    @content = params[:content]
+    @is_delayed = params[:is_delayed]
+    @pub_date = params[:pub_date]
+    @picture = params[:picture]
+  end
+
+  def submit
+    raise NotImplementedError, 'No user, content or picture provided' unless @user && (@content || @picture)
+
+    DelayMicropostWorker.perform_at(micropost_pub_date, user_id: @user.id, content: @content)
```

Comment:
I think, if delayed post creation is not required then you can create it right here without a sidekiq. Better to call it when it actually needed and keep your queue as short as possible for better performance

Reply:
thank you! fixed here https://github.com/saparjohnick/twitter_like/pull/1/commits/a7f2d7d827b7b638449b1488309662368d2bde42

Reply:
sorry for that, seems I missed one case with delaying -- fixed here (what do you think about this realisation with more methods?) https://github.com/saparjohnick/twitter_like/pull/1/commits/60338f8e03fbde22a635aed48d4d0ce50004dc51

also It was kind of tricky to deal with `if @is_delayed` because it was somehow lead to true condition branch for `is_delayed = "0"` But it works with `if @pub_date && @is_delayed`

Reply:
@saparjohnick I think it's because the `is_delayed` value passed directly from user input. Multiple methods are fine here.